worlds/tunic/__init__.py

```diff
@@ -307,8 +332,53 @@ def remove_filler(amount: int) -> None:
             if tunic_item.name in slot_data_item_names:
                 self.slot_data_items.append(tunic_item)
 
+        # pull out the filler so that we can place it manually during pre_fill
+        self.local_filler = []
+        if self.options.grass_randomizer and self.options.local_fill > 0 and self.multiworld.players > 1:
+            # skip items marked local or non-local, let fill deal with them in its own way
+            # discard grass from non_local if it's meant to be limited
+            if self.settings.limit_grass_rando:
+                self.options.non_local_items.value.discard("Grass")
+            all_filler = []
+            non_filler = []
+            amount_to_local_fill = int(self.options.local_fill.value * len(all_filler) / 100)
+            for item in tunic_items:
+                if item.classification in [ItemClassification.filler, ItemClassification.trap] and item.name not in self.options.local_items and item.name not in self.options.non_local_items:
+                    if len(self.local_filler) < amount_to_local_fill:
+                        self.local_filler.append(item)
+                    else:
+                        all_filler.append(item)
+                else:
+                    non_filler.append(item)
```

Comment:
Adds typing to `all_filler` and `non_filler`, changes the spacing of the `if` block and changes `item` which as already used earlier as a string to `tunic_item` which can properly be a TunicItem
```suggestion
            all_filler: List[TunicItem] = []
            non_filler: List[TunicItem] = []
            amount_to_local_fill = int(self.options.local_fill.value * len(all_filler) / 100)
            for tunic_item in tunic_items:
                if (tunic_item.classification in [ItemClassification.filler, ItemClassification.trap]
                        and tunic_item.name not in self.options.local_items
                        and tunic_item.name not in self.options.non_local_items):
                    if len(self.local_filler) < amount_to_local_fill:
                        self.local_filler.append(tunic_item)
                    else:
                        all_filler.append(tunic_item)
                else:
                    non_filler.append(tunic_item)
```